python/source.py

```diff
@@ -583,3 +582,12 @@ def beam_w0(self):
     @property
     def beam_E0(self):
         return self._beam_E0
+
+class IndexedSource(Source):
+    def __init__(self, src, num_pts, srcdata, amp_arr):
+        self.src = src
+        self.num_pts = num_pts
+        self.srcdata = srcdata
```

Comment:
Is `srcdata` a `meep::srcdata*` or something?  Would be good to have a comment here.